### DISCUSSION TITLE
Help getting a 4 ch RS485 to USB adaptor working with dbus-serialbattery

### DISCUSSION THREAD

**Discussion by shanetith** [Louisvdw/dbus-serialbattery#922] [Q&A]
I use a Waveshare USB to RS485 adaptor (https://www.waveshare.com/usb-to-rs485.htm) to read my Renogy batteries BMS, it works fine.  The problem is I have 6 batteries I need to read so I got 2 Waveshare USB to 4ch RS485 adaptors (https://www.waveshare.com/product/iot-communication/wired-comm-converter/usb-to-rs232-uart-fifo/usb-to-4ch-rs485.htm) but It will not read the BMS's.  I have tried reading them single and multiple but it will not work.  Anyone have any ideas how I might get them to work?

**Comment by kommando828** [comment]
The correct way to do this under RS485 Modbus RTU is to use the single USB to RS485 and daisy chain connect to all the BMS's.

Here is the scheme for SMA RS485 inverters in attached pic. You need a 120ohm resistor termination at each end

But first you need to change the ID's for the BMS's, so make then 2.3.4.5.6 and 7 so the Pi knows which BMS is which. If you cannot set the ID's then it cannot work. ID 1 is for the Pi running dbus-serial.

<img width="700" alt="SMA wiring CC CCA 2" src="https://github.com/Louisvdw/dbus-serialbattery/assets/32205016/608e09b8-56c1-4201-825d-3143517ecca8">

Take your current working setup and add one more BMS and see if it works.

**Reply by shanetith** [reply]
Thank you, I tried daisy chaining them and it didnt work,  just froze everything, but I didnt terminate or number them.  I will try that.  Any suggestions as to how I can find the numbering and try to change it.

**Comment by kommando828** [comment]
That is a question for Renogy, I quickly scanned one of their installation guides and did not see how to do it. You would normally have some dip switches or a RS485 writable register that you send the ID to, so you need the RS485 register map for your battery. If you have to write to a register then this tool is very good.

 https://github.com/SciFiDryer/ModbusMechanic

That as soon as you added the second battery the single RS485/USB froze suggests it was because both batteries had the same ID, both would have answered at the same time.

The installation guide I read said max number of batteries was 4 which may be an ID restriction.

**Reply by shanetith** [reply]
Thank you, I will look into this.
